src/App.tsx

```diff
@@ -15,48 +16,111 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortBy {
+  sortByAlphabet = 'sortByAlphabet',
+  sortByLength = 'sortByLength',
+  initState = '',
+}
+
+const sortGoods = (sortBy: SortBy) => {
+  const copy = [...goodsFromServer];
+
+  switch (sortBy) {
+    case SortBy.sortByAlphabet:
+      return copy.sort((good1, good2) => good1.localeCompare(good2));
+
+    case SortBy.sortByLength:
+      return copy.sort((good1, good2) => good1.length - good2.length);
+
+    case SortBy.initState:
+      return copy;
+
+    default:
+      return 0;
+  }
+};
+
+export const App = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortBy, setSortBy] = useState<SortBy>(SortBy.initState);
+  const [goods, setGoods] = useState([...goodsFromServer]);
```

Comment:
unnecessary state, you can use simple variable and return the value from `sortGoods` function